service/hook/gogs/gogs.go

```diff
@@ -86,13 +91,49 @@ func transformCodePushEvent(codePushEvent CodePushEventModel) hookCommon.Transfo
 		}
 	}
 
+	if len(lastCommit.CommitHash) == 0 {
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Missing commit hash"),
+		}
+	}
+
+	if !strings.HasPrefix(pushEvent.Ref, "refs/heads/") {
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Ref (%s) is not a head ref", pushEvent.Ref),
+		}
+	}
+
+	branch := strings.TrimPrefix(pushEvent.Ref, "refs/heads/")
+
 	return hookCommon.TransformResultModel{
 		TriggerAPIParams: []bitriseapi.TriggerAPIParamsModel{
 			{
 				BuildParams: bitriseapi.BuildParamsModel{
+					Branch:        branch,
 					CommitHash:    lastCommit.CommitHash,
 					CommitMessage: lastCommit.CommitMessage,
-					Branch:        branch,
+				},
+			},
+		},
+	}
+}
+
+func transformCreateEvent(createEvent CreateEventModel) hookCommon.TransformResultModel {
+	// Currently, we only support tag creation, not branches. Even if we wanted branch creation
```

Comment:
It's fine, in general we don't suggest doing builds for branch creation (when a branch is created with 0 changes) only when a push also happens on it, as that seems to fit the usual flow/what people expect (they don't really expect an "empty" branch creation to kick off a build, only after they push something to the new branch).